File: src/main/java/com/piveguyz/empickbackend/orgstructure/attendance/command/application/dto/AttendanceRecordCreateRequestDTO.java
package com.piveguyz.empickbackend.orgstructure.attendance.command.application.dto;

import lombok.*;

import java.time.LocalDateTime;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class AttendanceRecordCreateRequestDTO {

    private Integer attendanceCategoryId;    // 근태 카테고리 ID (0=출근, 1=퇴근, 2=지각, 3=조퇴)
    private LocalDateTime recordTime;        // 기록 시각 (null이면 현재 시각 사용)

    // memberId는 JWT에서 추출하므로 DTO에 포함하지 않음
}

File: Empick-FE/src/dto/attendance/attendanceRecordCreateRequestDTO.js
export default class AttendanceRecordCreateRequestDTO {
    constructor(attendanceCategoryId = null, recordTime = null) {
        this.attendanceCategoryId = attendanceCategoryId;    // 근태 카테고리 ID (1=출근, 2=퇴근, 3=지각, 4=조퇴)
        this.recordTime = recordTime;                        // 기록 시간
    }

    // memberId는 JWT에서 추출하므로 DTO에 포함하지 않음
}